app/src/main/java/com/brandon/forzenbook/viewmodels/ForzenTopLevelViewModel.kt

```diff
@@ -7,100 +7,97 @@ import android.os.Build
     val state: MutableState<LoginUiState>
         get() = _state
 
     sealed interface LoginUiState {
 
-        object PreCode : LoginUiState
+        data class PreCode(
+            val isEmailError: Boolean = false,
+            val isGenericError: Boolean = false,
+        ): LoginUiState
+
+        object CodeSent: LoginUiState
 
         data class Error(
-            val isGenericError: Boolean = false,
             val isEmailError: Boolean = false,
-            val isCodeError: Boolean = false,
-            val isInvalidCredentialsError: Boolean = false,
-            val isNetworkError: Boolean = false,
-        ) : LoginUiState
+            val isGenericError: Boolean = false,
+        ): LoginUiState
 
         object Loading : LoginUiState
 
-        data class PostCodeSent(
+        data class LoginWithCode(
             val loginToken: String,
         ) : LoginUiState
 
     }
 
-    fun dismissNotification() {
-        state.value = LoginUiState.PreCode
+    fun getCode(email: String) {
+        viewModelScope.launch {
+            state.value = LoginUiState.Loading
+            delay(5000)
+            if (email.isNotEmpty()) {
+                state.value = LoginUiState.PreCode(isEmailError = true)
+                cancel()
+            }
+            if (getLoginCodeUseCase(email)) {
+                state.value = LoginUiState.CodeSent
+            } else {
+                state.value = LoginUiState.PreCode(isGenericError = true)
+            }
+            Log.e("Brandon_Test_ViewModel","${state.value}")
+        }
     }
 
-    fun loginClicked(userName: String, code: String) {
+    fun loginClicked(email: String, code: String) {
         viewModelScope.launch {
             state.value = LoginUiState.Loading
-            loginUserName = userName
-            logincode = code
             delay(5000)
-            if (userName.isEmpty() || code.isEmpty()) {
-                state.value = LoginUiState.Error(
-                    isInvalidCredentialsError = true,
-                )
+            if (email.isEmpty() || code.isEmpty()) {
+                state.value = LoginUiState.PreCode(isEmailError = true)
                 cancel()
             }
-            val token = loginUseCase(userName, code)
+            val token = loginUseCase(email, code)
             if (token != null) {
-                if (token.token != null) {
-                    state.value = LoginUiState.PostCodeSent(loginToken = token.token)
-                } else {
-                    state.value = LoginUiState.Error(isGenericError = true)
-                }
+                    if (token.token != null) {
+                        state.value = LoginUiState.LoginWithCode(loginToken = token.token)
+                    }
+                    else { state.value = LoginUiState.Error(isGenericError = true) }
             }
-            Log.e(VIEWMODEL_ERROR_TAG, "${state.value}")
+            Log.e("Brandon_Test_ViewModel","${state.value}")
```

Comment:
This line isn't formatted correctly. You dare invoke the wrath of KTLint?

Reply:
Fixed it with a ctrlS